### ISSUE TITLE
Accessibility: aria-labelledby and aria-label

### ISSUE THREAD

**Issue by AdrianHostettler** [snapappointments/bootstrap-select#2294]
Hy, I hope you are doing fine!
(This is a feature request)

I was wondering if you could add new optional attributes concerning labels, so a screen reader would read an assigned label.

e.g. **data-aria-label** and **data-aria-labelledby**

Let's say i have a structure like this:

```
<label id="Fruits_label" for="Fruits">This is an awesome snapappointments Fruits Select Box</label>
<select id="Fruits" class="selectpicker" aria-labelledby="Fruits_label">.....</select>

```
Then the visible select box results in the <button> element, that is rendered as code after the select element. The select element itself is hidden.
Of course the generated button element can not have the same id as the hidden select element, so the "for" attribute of the label element can not work. 

But it could be possible to add following attributes to the button element, so it wouldn't matter, that the "for" attribute doesn't link correctly: 

**aria-labelledby:**
```
<label id="Fruits_label" for="Fruits">This is an awesome snapappointments Fruits Select Box</label>
<select id="Fruits" class="selectpicker" aria-labelledby="Fruits_label">.....</select>
<button type="button" class="btn dropdown-toggle btn-secondary" data-toggle="dropdown" role="combobox" ..... **aria-labelledby="Fruits_label"**></button>
```

And if one does want to provide the label directly as text for screenreaders, and not through id of an existing label:

**aria-label:**
```
<label id="Fruits_label" for="Fruits">This is an awesome snapappointments Fruits Select Box</label>
<select id="Fruits" class="selectpicker" aria-labelledby="Fruits_label">.....</select>
<button type="button" class="btn dropdown-toggle btn-secondary" data-toggle="dropdown" role="combobox" ..... **aria-label="This is an awesome snapappointments Fruits Select Box"**></button>
```

This would make the intended usage of the selectbox much more clearer for screenreader users.

If you'd like a pull request, let me know, but i think its probably done easily without? 

With kind regards

A

**Comment by AdrianHostettler** [comment]
As an addition, i found following pull-request, that probably solves the issue already partially:

https://github.com/snapappointments/bootstrap-select/pull/1882/files

The 2 new data-attributes are probably unnecessary, if you can just take the values from the select element, directly to the button element.

aria-labelledby="'+this.$element.attr("aria-labelledby")

Would that be possible?

With kind regards

**Comment by rom1vtle** [comment]
Yes, thank you, i also need this for accessibility

**Comment by caramori** [comment]
I also caught that in my test for screen readers. Someone with visual impairment would only get the 'Combobox collapsed' audio when focusing on the element. I tried aria-label and aria-labelledby or even title and none are currently being used.

**Comment by ghost** [comment]
Im using this function to pass the aria-label of the original select element to the button element generated by bootstrap-select, seems to work fine with screen readers: 

```
function fixSelectpickerAriaLabels() {
	$('.selectpicker').each(function() {
		var button = $('button[data-id="'+this.id+'"]').first();
		if(button.length > 0) {
			button.attr('aria-label', $(this).attr('aria-label'));
		}
	});
}
```

This snippet depends on jquery and should be executed after document ready event.

**Comment by simobari** [comment]
Hello, 
I have experienced the same problem. Do you have any plan to fix this?

Thanks
